static/js/message_finder.js

```diff
@@ -0,0 +1,49 @@
+import {all_messages_data} from "./all_messages_data";
+import * as message_store from "./message_store";
+
+/*
+    The functions in this module are highly deprecated.
+
+    We should have their callers do a better job
+    of tracking messages by topic internally.
+*/
+
+export function get_messages_in_topic(stream_id, topic) {
+    return all_messages_data
+        .all_messages()
+        .filter(
+            (x) =>
+                x.type === "stream" &&
+                x.stream_id === stream_id &&
+                x.topic.toLowerCase() === topic.toLowerCase(),
+        );
+}
+
+export function get_max_message_id_in_stream(stream_id) {
+    let max_message_id = 0;
+    for (const msg of all_messages_data.all_messages()) {
+        if (msg.type === "stream" && msg.stream_id === stream_id && msg.id > max_message_id) {
+            max_message_id = msg.id;
+        }
+    }
+    return max_message_id;
+}
```

Comment:
I think these two could actually just go in `all_messages_data.js`?  They're pure functions that just process that data set.

Reply:
I considered putting them there, but then they complexify any module that just wants to import `all_messages_data`.  As I mention in the commit message, the end game for these functions is either to eliminate them or put them inside of `recent_topics_data` (once we split them out).

Reply:
> complexify any module

By this, do you mean complicate the way they do the import?

Reply:
I'm not sure I follow -- wouldn't `const {all_messages_data} = zrequire("all_messages_data");` be the same for everything but the modules that need to import these functions?

Reply:
I am saying that any module that imports `all_messages_data` would now have the additional transitive dependency of `message_store`, plus anybody reading the code who just cares about `all_messages_data` will see these deprecated functions.

As I mentioned in the commit message, the end game for these functions is to either upstream them to `recent_topics_data` or just remove them completely (by making the callers smarter).  This is just a tactical change.  We can always put them into `all_messages_data` later if my gut is wrong is here.

Reply:
That's true only for the 3rd function below this, which calls into `message_store`.  The two functions above don't have any dependencies, and can be just moved to `all_messages_data`.